dbms/src/Functions/LeastGreatest.h

```diff
@@ -120,5 +121,363 @@ class FunctionVectorizedLeastGreatest : public IFunction
+    explicit FunctionLeastGreatestString() = default;
+
+    static FunctionPtr create(const Context & context [[maybe_unused]])
+    {
+        return std::make_shared<FunctionLeastGreatestString>();
+    }
+
+    String getName() const override { return name; }
+    bool isVariadic() const override { return true; }
+
+    void setCollator(const TiDB::TiDBCollatorPtr & collator_) override { collator = collator_; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+    bool useDefaultImplementationForNulls() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() < 2)
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be at least 2.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        for (const auto & argument : arguments)
+        {
+            if (!argument->isString())
+            {
+                throw Exception(
+                    fmt::format("argument type not string"),
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+        }
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        size_t num_arguments = arguments.size();
+        if (num_arguments < 2)
+        {
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be at least 2.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        }
+        if (collator)
+            executeInternal<true>(block, arguments, result);
+        else
+            executeInternal<false>(block, arguments, result);
+    }
+    template <bool use_collator>
+    void executeInternal(Block & block, const ColumnNumbers & arguments, size_t result) const
+    {
+        size_t num_arguments = arguments.size();
+
+        using impl = StringOperationImpl<least, use_collator>;
+
+        std::vector<const ColumnConst *> const_columns;
+        std::vector<const ColumnString *> string_columns;
+        for (size_t i = 0; i < num_arguments; ++i)
+        {
+            const auto * c = block.getByPosition(arguments[i]).column.get();
+            const auto * c_string = checkAndGetColumn<ColumnString>(c);
+            const ColumnConst * c_const = checkAndGetColumnConstStringOrFixedString(c);
+            if (c_const)
+                const_columns.emplace_back(c_const);
+            if (c_string)
+                string_columns.emplace_back(c_string);
+        }
+
+        String const_res;
+        // for const columns
+        if (!const_columns.empty())
+        {
+            if (const_columns.size() == 1)
+            {
+                const_res = const_columns[0]->getValue<String>();
```

Comment:
I think `getValue<String>` is slower than `getDataAt(0).toString()` because it's heavy to construct `Field`.